Modules/rating/Controller/RatingController.php

```diff
@@ -0,0 +1,348 @@
+        //             $sql = 'SELECT bk_calendar_entry.resource_id , core_users.id as user_id, core_users.login as user_login, core_users.email as user_email, re_info.name as resource_name FROM bk_calendar_entry
+
+        $resources = [];
+        foreach($bkentries as $entry) {
+            if(isset($entry['resource_id'])) {  
+                $resources[$entry['resource_id']] = [
+                    'vid' => intval($entry['resource_id']),
+                    'id' => 0,
+                    'module' => 'booking',
+                    'name' => $entry['resource_name'],
+                    'rate' => 0,
+                    'comment' => null,
+                    'anon' => 1,
+                ];
+            }
+        }
+
+        // TODO add projects
+        // closedProjectsByPeriod
+        $s = new SeProject();
+        $closed_projects = $s->closedProjectsByPeriod($id_space, $_SESSION['id_user'], date('Y-m-d', $campaign['from_date']), date('Y-m-d', $campaign['to_date']));
+        $projects = [];
+        foreach($closed_projects as $entry) {
+                $projects[$entry['id']] = [
+                    'vid' => intval($entry['id']),
+                    'id' => 0,
+                    'module' => 'projects',
+                    'name' => $entry['name'],
+                    'rate' => 0,
+                    'comment' => null,
+                    'anon' => 1,
+                ];
+        }
+
+        $r = new Rating();
+        $stats = $r->list($id_space, campaign: $campaign['id']);
+        foreach($stats as $stat){
+            if($stat['module'] == 'booking' && isset($resources[$stat['resource']])) {
+                    $resources[$stat['resource']]['id'] = $stat['id'];
+                    $resources[$stat['resource']]['rate'] = round($stat['rate']);
+                    $resources[$stat['resource']]['comment'] = $stat['comment'];
+                    $resources[$stat['resource']]['anon'] = $stat['anon'];
+                }
+            if($stat['module'] == 'projects' && isset($projects[$stat['resource']])) {
+                $projects[$stat['resource']]['id'] = $stat['id'];
+                $projects[$stat['resource']]['rate'] = round($stat['rate']);
+                $projects[$stat['resource']]['comment'] = $stat['comment'];
+                $projects[$stat['resource']]['anon'] = $stat['anon'];
+            }
+        }
+        return $this->render(['lang' => $lang, 'campaign' => $campaign, 'data' => ['resources' => $resources, 'projects' => $projects]]);
+    }
+
+
+
+    public function rateAction($id_space, $id_campaign) {
+        $plan = new CorePlan($this->currentSpace['plan'], $this->currentSpace['plan_expire']);
+        if(!$plan->hasFlag(CorePlan::FLAGS_SATISFACTION)) {
+            throw new PfmAuthException('Sorry, space does not have this feature plan');
+        }
+        $userSpaceStatus = $this->getUserSpaceStatus($id_space, $_SESSION["id_user"]);
+        if($userSpaceStatus != CoreSpace::$USER) {
```

Comment:
Space managers or admins status can't evaluate ? So maybe we shouldn't send them campaign emails... And maybe allow campaign author to choose to include managers and / or admins

Reply:
there is no sens for managers/admins to  evaluate. Satisfaction survey is for the users/customers. You don't give yourself a note....

but true, email is sent to users with a booking,... which could be a manager/admin. Their role should be checked for unnecessary emails

Reply:
Sometimes, "users" can have manager status. But it results from a wrong usage of PFM. ok

Reply:
Let's keep it like this for the moment, we will see on usage (if any...)